### ISSUE TITLE
Issue with Client Addons

### ISSUE THREAD

**Issue by timothypgny** [auth0/terraform-provider-auth0#757]
### Checklist

- [X] I have looked into the [README](https://github.com/auth0/terraform-provider-auth0#readme) and have not found a suitable solution or answer.
- [X] I have looked into the [documentation](https://registry.terraform.io/providers/auth0/auth0/latest/docs) and have not found a suitable solution or answer.
- [X] I have searched the [issues](https://github.com/auth0/terraform-provider-auth0/issues) and have not found a suitable solution or answer.
- [X] I have upgraded to the [latest version](https://github.com/auth0/terraform-provider-auth0/releases/latest) of this provider and the issue still persists.
- [X] I have searched the [Auth0 Community](https://community.auth0.com/c/sdks/5) forums and have not found a suitable solution or answer.
- [X] I agree to the terms within the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md).

### Description

We are currently using version 0.40.0 of the auth0 provider and trying to use the beta version (and eventually GA version) so we can upgrade our actions to use Node 18 instead of Node 16.

We have a client:

We use python to interact with the auth0 provider and seeing an issue when trying to create a regular web application with SAML2 web app addon:

from imports.auth0 import Client
from imports.auth0 import ClientAddons

        client = Client(
            self,
            name="Application",
            id_="auth0_application",
            description=" Application",
            app_type="regular_web",
            callbacks=["https://foo/saml/sso"],
            addons=ClientAddons(
                samlp={
                    "create_upn_claim": False,
                    "passthrough_claims_with_no_mapping": False,
                    "map_identities": False,
                    "name_identifier_format": "rn:oasis:names:tc:SAML:2.0:nameid-format:transient",
                }
            ),
        )

Using 0.40.0, it is working as expected:

![image](https://github.com/auth0/terraform-provider-auth0/assets/93941231/967aec6b-c783-4578-92da-a6716421e112)

and the SAML properties are set up properly:

![image](https://github.com/auth0/terraform-provider-auth0/assets/93941231/563d1994-8be1-4377-bcfd-60ef0447b26b)

However in the beta version, it has more addons created:

![image](https://github.com/auth0/terraform-provider-auth0/assets/93941231/2eeed002-372f-44bc-b3a0-ba9f3b62767b)

and the SAML properties are not set up properly:

![image](https://github.com/auth0/terraform-provider-auth0/assets/93941231/c8ac4811-c1ad-4501-a849-5ceb0aeb8e3b)

I tried to make the changes to the SAML properties to match what was working before but even those are not getting picked up.  

We have a rule that maps additional SAML attributes and it seems like that's not getting fired in the beta version, but works everywhere else.

When I create the web application manually, everything works as expected.

It also can't read this application in the current tfstate generated by the GA version.  We can't delete this application as the SAML metadata has the client id in it.

### Expectation

It should work as GA version.

### Reproduction

from imports.auth0 import Client
from imports.auth0 import ClientAddons

        client = Client(
            self,
            name="Application",
            id_="auth0_application",
            description=" Application",
            app_type="regular_web",
            callbacks=["https://foo/saml/sso"],
            addons=ClientAddons(
                samlp={
                    "create_upn_claim": False,
                    "passthrough_claims_with_no_mapping": False,
                    "map_identities": False,
                    "name_identifier_format": "rn:oasis:names:tc:SAML:2.0:nameid-format:transient",
                }
            ),
        )

Then run cdktf deploy and check generated web application in Auth0 dashboard.

### Auth0 Terraform Provider version

1.0.0-beta.0

### Terraform version

v1.2.3

**Comment by sergiught** [comment]
Thanks for opening this dedicated issue @timothypgny 🙏🏻, the info you've provided is great and I'll work on a fix asap. We'll look into releasing the fix in a follow up v1.0.0-beta.1 release hopefully by the end of this week or the beginning of the next one. Appreciate your patience and awesome work on testing the beta already 💪🏻

**Comment by sergiught** [comment]
Hey @timothypgny, @renan-abm 👋🏻 

We've followed up with a fix for this in https://registry.terraform.io/providers/auth0/auth0/1.0.0-beta.1. Please make sure you update the version in your terraform configs:

```
terraform {
  required_providers {
    auth0 = {
      source  = "auth0/auth0"
       version = "1.0.0-beta.1"
    }
  }
}
```

If you're upgrading and are already managing an existing `auth0_client` resource please check the migration guide: https://github.com/auth0/terraform-provider-auth0/blob/main/MIGRATION_GUIDE.md#client-addons, specifically before upgrading, run a `terraform state rm auth0_client.<resource_name>.` Then upgrade to `v1.0.0-beta.1` and re-import the client resource using `terraform import auth0_client.<resource_name> <client_id>`. This is needed due to the fact that the addon types have been better typed, thus changing as well how we store these settings within the state.

Let us know if you have any other issues! Much appreciated again for the beta feedback! 🙇🏻
